services/openvpn/service/factory.go

```diff
@@ -47,10 +55,24 @@ func NewManager(nodeOptions node.Options,
 	mapPort func(int) (releasePortMapping func()),
 	natEventGetter NATEventGetter,
 	portPool port.ServicePortSupplier,
+	publisher publisher,
 ) *Manager {
-	sessionValidator := openvpn_session.NewValidator(sessionMap, identity.NewExtractor())
+	clientMap := openvpn_session.NewClientMap(sessionMap)
+
+	sessionValidator := openvpn_session.NewValidator(clientMap, identity.NewExtractor())
+
+	callback := func(sbc bytecount.SessionByteCount) {
+		sessions := clientMap.GetClientSessions(sbc.ClientID)
+		if len(sessions) > 0 {
+			publisher.Publish(event.DataTransfered, event.DataTransferEventPayload{
+				ID:   string(sessions[0]),
```

Comment:
Bit awkward here, I understand that in our env single clientID should have at most 1 active session, but we get a first session only here, thus logic here implies that we might have more then one.  Actually i don't know if there might be any situation when this same clientID would have more than 1 session. We probably should fail early if we see that we are adding another session for the same clientID in GetClientSessions.

Reply:
i've added a log here, don't think we need to fail here as this situation should not be possible under normal use. Either way - the whole client map thing could use a second look at it when we have the time.